openmls/src/group/tests/test_commit_validation.rs

```diff
@@ -226,217 +251,172 @@ fn test_valsem200(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
-
-    // Now do the remove
-    // Clear the pending commit.
-    alice_group.clear_pending_commit();
-
-    // Before we can test the commit, we first have to add Charlie so we can
-    // actually remove someone and have someone else process the commit.
-    let charlie_credential = generate_credential_bundle(
-        "Charlie".into(),
-        CredentialType::Basic,
-        ciphersuite.signature_algorithm(),
-        backend,
-    )
-    .expect("An unexpected error occurred.");
-
-    // Generate KeyPackages
-    let charlie_key_package =
-        generate_key_package_bundle(&[ciphersuite], &charlie_credential, vec![], backend)
-            .expect("An unexpected error occurred.");
-
-    let (_msg_out, welcome) = alice_group
-        .add_members(backend, &[charlie_key_package])
-        .expect("error adding charlie");
-
-    alice_group
-        .merge_pending_commit()
-        .expect("error merging pending commit");
-
-    let mls_group_config = MlsGroupConfig::builder()
-        .wire_format_policy(wire_format_policy)
-        .build();
-
-    let mut charlie_group = MlsGroup::new_from_welcome(
-        backend,
-        &mls_group_config,
-        welcome,
-        Some(alice_group.export_ratchet_tree()),
-    )
-    .expect("Error creating group.");
-
-    // Create the remove.
-    let serialized_remove = alice_group
-        .remove_members(backend, &[bob_group.own_leaf_index()])
-        .expect("Error creating remove")
-        .tls_serialize_detached()
-        .expect("Could not serialize message.");
-
-    let remove_message_in = erase_path(backend, &serialized_remove, &alice_group);
-
-    // Have Charlie process everything
-    let unverified_message = charlie_group
-        .parse_message(remove_message_in, backend)
-        .expect("Could not parse message.");
-
-    let err = charlie_group
-        .process_unverified_message(unverified_message, None, backend)
-        .expect_err("Could process unverified message despite missing path.");
-
-    assert_eq!(
-        err,
-        UnverifiedMessageError::InvalidCommit(StageCommitError::RequiredPathNotFound)
-    );
-
-    let original_remove_plaintext =
-        VerifiableMlsAuthContent::tls_deserialize(&mut serialized_remove.as_slice())
-            .expect("Could not deserialize message.");
-
-    // Positive case
-    let unverified_message = charlie_group
-        .parse_message(MlsMessageIn::from(original_remove_plaintext), backend)
-        .expect("Could not parse message.");
-    if let ProcessedMessage::StagedCommitMessage(staged_commit) = charlie_group
-        .process_unverified_message(unverified_message, None, backend)
-        .expect("Unexpected error.")
-    {
-        charlie_group
-            .merge_staged_commit(*staged_commit)
-            .expect("error merging staged commit");
-    } else {
-        panic!("Expected a staged commit.");
+    let add_proposal = || {
+        let debbie_credential = generate_credential_bundle(
```

Comment:
The traditional fourth participant is called Dave, but Debbie works, too 😄.

Reply:
Let's go for Dave 😆